presentation/src/main/java/com/anytypeio/anytype/presentation/multiplayer/ShareSpaceViewModel.kt

```diff
@@ -48,16 +56,67 @@ class ShareSpaceViewModel(
     val isCurrentUserOwner = MutableStateFlow(false)
 
     init {
-        proceedWithGeneratingInviteLink()
+        proceedWithSpaceAccessTypeSubscription()
         proceedWithSpaceMemberSubscription()
     }
 
+    private fun proceedWithSpaceAccessTypeSubscription() {
+        viewModelScope.launch {
+            container.subscribe(
+                StoreSearchParams(
+                    subscription = SHARE_SPACE_SPACE_SUBSCRIPTION,
+                    keys = buildList {
+                        add(Relations.ID)
+                        add(Relations.SPACE_ACCESS_TYPE)
+                        add(Relations.TARGET_SPACE_ID)
+                    },
+                    limit = 1,
+                    filters = buildList {
+                        add(
+                            DVFilter(
+                                relation = Relations.TARGET_SPACE_ID,
+                                value = params.space.id,
+                                condition = DVFilterCondition.EQUAL
+                            )
+                        )
+                    }
+                )
+            ).mapNotNull { results ->
+                val space = results.firstOrNull()
+                if (space != null) {
+                    val wrapper = ObjectWrapper.SpaceView(space.map)
+                    when(wrapper.spaceAccessType) {
+                        SpaceAccessType.PRIVATE -> {
+                            ShareLinkViewState.NotGenerated
+                        }
+                        SpaceAccessType.SHARED -> {
+                            val link = getSpaceInviteLink.async(params.space)
+                            if (link.isSuccess) {
+                                ShareLinkViewState.Share(link.getOrThrow().scheme)
+                            } else {
+                                null
+                            }
+                        }
+                        else -> {
+                            null
+                        }
+                    }
+                } else {
+                    null
+                }
+
+            }.collect { result ->
```

Comment:
.catch { Timber.e(it, "Error while subscribing to ...") }

Reply:
done